Sources/SQLKit/Builders/Implementations/SQLInsertBuilder.swift

```diff
@@ -0,0 +1,384 @@
+    ///
+    /// Example usage:
+    ///
+    /// ```swift
+    /// let earth = Planet(id: nil, name: "Earth", isInhabited: true)
+    /// let mars = Planet(id: nil, name: "Mars", isInhabited: false)
+    ///
+    /// try await sqlDatabase.insert(into: "planets")
+    ///     .models([earth, mars], keyEncodingStrategy: .convertToSnakeCase)
+    ///     .run()
+    ///
+    /// // Effectively the same as:
+    /// try await sqlDatabase.insert(into: "planets")
+    ///     .columns("id", "name", "is_inhabited")
+    ///     .values(SQLBind(earth.id), SQLBind(earth.name), SQLBind(earth.isInhabited))
+    ///     .values(SQLBind(mars.id), SQLBind(mars.name), SQLBind(mars.isInhabited))
+    ///     .run()
+    /// ```
+    ///
+    /// > Note: The term "model" does _not_ refer to Fluent's `Model` type. Fluent models are not compatible with
+    /// > this method or any of its variants.
+    ///
+    /// - Parameters:
+    ///   - models: Array of values of a given type to insert. The given type may be any encodable type which
+    ///     represents an aggregate value.
+    ///   - prefix: See ``SQLQueryEncoder/prefix``.
+    ///   - keyEncodingStrategy: See ``SQLQueryEncoder/keyEncodingStrategy-swift.property``.
+    ///   - nilEncodingStrategy: See ``SQLQueryEncoder/nilEncodingStrategy-swift.property`.
+    ///   - userInfo: See ``SQLQueryEncoder/userInfo``.
+    @inlinable
+    @discardableResult
+    public func models(
+        _ models: [some Encodable],
+        prefix: String? = nil,
+        keyEncodingStrategy: SQLQueryEncoder.KeyEncodingStrategy = .useDefaultKeys,
+        nilEncodingStrategy: SQLQueryEncoder.NilEncodingStrategy = .default,
+        userInfo: [CodingUserInfoKey: any Sendable] = [:]
+    ) throws -> Self {
+        try self.models(models, with: .init(prefix: prefix, keyEncodingStrategy: keyEncodingStrategy, nilEncodingStrategy: nilEncodingStrategy, userInfo: userInfo))
+    }
+    
+    /// Use an array of `Encodable` values to generate rows to insert and add those rows to the query.
+    ///
+    /// Example usage:
+    /// ```swift
+    /// let earth = Planet(id: nil, name: "Earth", isInhabited: true)
+    /// let mars = Planet(id: nil, name: "Mars", isInhabited: false)
+    /// let encoder = SQLQueryEncoder(nilEncodingStrategy: .asNil)
+    ///
+    /// try await sqlDatabase.insert(into: "planets")
+    ///     .models([earth, mars], with: encoder)
+    ///     .run()
+    ///
+    /// // Effectively the same as:
+    /// try await sqlDatabase.insert(into: "planets")
+    ///     .columns("id", "name", "isInhabited")
+    ///     .values(SQLLiteral.null, SQLBind(earth.name), SQLBind(earth.isInhabited))
+    ///     .values(SQLLiteral.null, SQLBind(mars.name), SQLBind(mars.isInhabited))
+    ///     .run()
+    /// ```
+    ///
+    /// > Note: The term "model" does _not_ refer to Fluent's `Model` type. Fluent models are not compatible with
```

Comment:
IMO this isn't needed, this package can be used without Fluent too so people who might do that could be confused